tldw_Server_API/app/core/DB_Management/backends/factory.py

```
@@ -46,6 +54,240 @@ def _describe_sqlite_target(config: DatabaseConfig) -> str:
+        parsed = _url.urlparse(raw_path)
+    except (TypeError, ValueError):
+        return False
+    query = _url.parse_qs(parsed.query or "")
+    mode = (query.get("mode", [""])[0] or "").lower()
+    cache = (query.get("cache", [""])[0] or "").lower()
+    if mode != "memory" or cache != "shared":
+        return False
+    return bool((parsed.path or "").strip("/"))
+
+
+def _normalize_sqlite_target(raw_path: str) -> str:
```

Ruff (0.15.9)

[warning] Too many return statements (8 > 6)

(PLR0911)